src/lib/mdns/Advertiser_ImplMinimalMdns.cpp

```diff
@@ -559,6 +561,149 @@ FullQName AdvertiserMinMdns::GetCommisioningTextEntries(const CommisioningAdvert
+                 .IsValid())
+        {
+            ChipLogError(Discovery, "Failed to add short discriminator PTR record mDNS responder");
+            return CHIP_ERROR_NO_MEMORY;
+        }
+    }
+
+    {
+        sprintf(nameBuffer, "L%04d", params.GetLongDiscriminator());
+        FullQName longServiceName = AllocateQName(nameBuffer, "_sub", "_chipd", "_udp", "local");
+        ReturnErrorCodeIf(longServiceName.nameCount == 0, CHIP_ERROR_NO_MEMORY);
+        if (!AddResponder<PtrResponder>(longServiceName, operationalServerName)
+                 .SetReportAdditional(operationalServerName)
+                 .SetReportInServiceListing(true)
+                 .IsValid())
+        {
+            ChipLogError(Discovery, "Failed to add long discriminator PTR record mDNS responder");
+            return CHIP_ERROR_NO_MEMORY;
+        }
+    }
+
+    if (params.GetVendorId().HasValue())
+    {
+        sprintf(nameBuffer, "V%d", params.GetVendorId().Value());
+        FullQName vendorServiceName = AllocateQName(nameBuffer, "_sub", "_chipd", "_udp", "local");
+        ReturnErrorCodeIf(vendorServiceName.nameCount == 0, CHIP_ERROR_NO_MEMORY);
+
+        if (!AddResponder<PtrResponder>(vendorServiceName, operationalServerName)
+                 .SetReportAdditional(operationalServerName)
+                 .SetReportInServiceListing(true)
+                 .IsValid())
+        {
+            ChipLogError(Discovery, "Failed to add vendor discriminator PTR record mDNS responder");
+            return CHIP_ERROR_NO_MEMORY;
+        }
+    }
+
+    if (!AddResponder<TxtResponder>(TxtResourceRecord(operationalServerName, GetCommisionableTextEntries(params)))
+             .SetReportAdditional(serverName)
+             .IsValid())
+    {
+        ChipLogError(Discovery, "Failed to add TXT record mDNS responder");
+        return CHIP_ERROR_NO_MEMORY;
+    }
+
+    ChipLogProgress(Discovery, "CHIP minimal mDNS configured as 'Commisionable device'.");
+
+    return CHIP_NO_ERROR;
+}
+
+FullQName AdvertiserMinMdns::GetCommisionableTextEntries(const CommisionableAdvertisingParameters & params)
+{
+    // a discriminator always exists
+    char txtDiscriminator[32];
+    sprintf(txtDiscriminator, "D=%d", params.GetLongDiscriminator());
+
+    if (!params.GetVendorId().HasValue())
+    {
+        return AllocateQName(txtDiscriminator);
+    }
+
+    // Need to also set a vid/pid string
+    char txtVidPid[64];
+    if (params.GetProductId().HasValue())
+    {
+        sprintf(txtVidPid, "V=%d+%d", params.GetVendorId().Value(), params.GetProductId().Value());
+    }
+    else
+    {
+        sprintf(txtVidPid, "V=%d", params.GetVendorId().Value());
+    }
+
+    char txtPairingInstrHint[128];
+    if (params.GetPairingHint().HasValue())
+    {
+        // sprintf(txtPairingInstrHint, "P=%d", params.GetPairingHint().Value());
+        sprintf(txtPairingInstrHint, "P=%s+%d", params.GetPairingInstr().Value(), params.GetPairingHint().Value());
```

Comment:
Need to ensure that GetPairingInstr() HasValue.

Reply:
Will do